proto/provenance/reward/v1/reward.proto

```diff
@@ -0,0 +1,254 @@
+  // ClaimStatus is the state a claim is in
+  enum ClaimStatus {
+    // undefined state
+    CLAIM_STATUS_UNSPECIFIED = 0;
+    // unclaimable status
+    CLAIM_STATUS_UNCLAIMABLE = 1;
+    // unclaimable claimable
+    CLAIM_STATUS_CLAIMABLE = 2;
+    // unclaimable claimed
+    CLAIM_STATUS_CLAIMED = 3;
+    // unclaimable expired
+    CLAIM_STATUS_EXPIRED = 4;
+  }
+  // The id of the reward program that this share belongs to.
+  uint64 reward_program_id = 1;
+  // The id of the claim period that the share belongs to.
+  uint64 claim_period_id = 2;
+  // Owner of the reward account state.
+  string address = 3 [(gogoproto.moretags) = "yaml:\"address\""];
+  // The number of actions done by this account, mapped by action type.
+  map<string, uint64> action_counter = 4;
+  // The amount of granted shares for the address in the reward program's claim period.
+  uint64 shares_earned = 5;
+  // The status of the claim.
+  ClaimStatus claim_status = 6;
+}
+
+// QualifyingAction can be one of many action types.
+message QualifyingAction {
+  option (gogoproto.equal)            = true;
+  option (gogoproto.goproto_stringer) = true;
+  // type of action to process
+  oneof type {
+    ActionDelegate delegate = 1;
+    ActionTransfer transfer = 2;
+    ActionVote     vote     = 3;
+  }
+}
+
+// QualifyingActions contains a list of QualifyingActions.
+message QualifyingActions {
+  // The actions that count towards the reward.
+  repeated QualifyingAction qualifying_actions = 1
+      [(gogoproto.nullable) = false, (gogoproto.moretags) = "yaml:\"qualifying_actions"];
+}
+
+// ActionDelegate represents the delegate action and its required eligibility criteria.
+message ActionDelegate {
+  option (gogoproto.equal)            = true;
+  option (gogoproto.goproto_stringer) = false;
+
+  // Minimum number of successful delegates.
+  uint64 minimum_actions = 1;
+  // Maximum number of successful delegates.
+  uint64 maximum_actions = 2;
+  // Minimum amount that the user must have currently delegated on the validator.
+  cosmos.base.v1beta1.Coin minimum_delegation_amount = 3;
+  // Maximum amount that the user must have currently delegated on the validator.
+  cosmos.base.v1beta1.Coin maximum_delegation_amount = 4;
+  // Minimum percentile that can be below the validator's power ranking.
+  string minimum_active_stake_percentile = 5
+      [(gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Dec", (gogoproto.nullable) = false];
+  // Maximum percentile that can be below the validator's power ranking.
+  string maximum_active_stake_percentile = 6
+      [(gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Dec", (gogoproto.nullable) = false];
+}
+
+// ActionTransfer represents the transfer action and its required eligibility criteria.
+message ActionTransfer {
+  option (gogoproto.equal)            = true;
+  option (gogoproto.goproto_stringer) = false;
+
+  // Minimum number of successful delegates.
```

Comment:
This comment needs to be updated for the transfer action, similar for the next several occurrences of copy/paste

Reply:
good catch, fixed to reflect transfer action.